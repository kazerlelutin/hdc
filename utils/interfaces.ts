export interface functionsEnabled {
    name: string,
    description: string,
    href: string
}