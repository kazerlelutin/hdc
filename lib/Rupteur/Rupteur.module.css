.container {
    cursor: pointer;
}

.rupteur {
    display: block;
    position: relative;
    border-radius: 25px;
    width: 35px;
    height: 20px;
    box-shadow: inset 1px 1px 5px black;
}

.rupteur[data-theme="dark"] {
    background: #31163b; 
}

.rupteur[data-theme="light"] {
    background: #3a67bb; 
}

.state, .state:before {
    position: absolute;
    width: 16px;
    height: 16px;
    border-radius: 50%;
    transition: 300ms ease-in-out;
}

.state {
    top: 2px;
    left: 2px;
    overflow: hidden;
}

.rupteur[data-theme="dark"] .state {
    background: #7a787a; 
    box-shadow: 0 0 5px rgba(255,255,255,.5);
}

.rupteur[data-theme="light"] .state {
    background: radial-gradient(rgb(235, 236, 155) 20%,rgb(206, 209, 4));
    box-shadow: 0 0 15px rgba(255,255,255,.6);
    left: calc(100% - 18px)
}

.rupteur[data-theme="dark"] .state:before{
    content: '';
    opacity: 1;
    left: 5px;
    top: 0;
    background: rgba(53, 50, 18, 0.5);
}

.rupteur[data-theme="light"] .state:before{
    content: '';
    opacity: 0;
    left: -20px;
    top: 0;
    background: rgba(53, 50, 18, 0.5);
}