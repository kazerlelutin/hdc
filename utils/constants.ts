export const PREFIX = "hdc";
export const DB_USER_PREFS = "hdc_db_prefs";
