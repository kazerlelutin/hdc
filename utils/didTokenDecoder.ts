import { Magic } from "@magic-sdk/admin";
import dayjs from "dayjs";

export default function didTokenDecoder(didToken: string): string {
  const mAdmin = new Magic(process.env.MAGIC_SECRET_KEY);
  try {
    const payload = mAdmin.token.decode(didToken);
    //console.log(payload,dayjs.unix(payload[1].ext).format('DD-MM-YY hh:mm:ss'))
    if (dayjs(dayjs.unix(payload[1].ext)).diff(dayjs()) > 0) {
      return payload[1].iss;
    } else {
      throw new Error("non valide");
    }
  } catch (e) {
    throw new Error(e);
  }
}
