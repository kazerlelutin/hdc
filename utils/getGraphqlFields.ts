import { GraphQlInfo } from './interfaces';

export default function getGraphqlFields(info: GraphQlInfo) {
  return info.fieldNodes[0].selectionSet.selections.map(
    (item) => item.name.value
  );
}
