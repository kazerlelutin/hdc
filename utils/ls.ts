export default class Ls {
  prefix = "hdc_ls_";
  userKey = this.prefix + "user_t";
  userToken: string;

  constructor() {
    this.userToken = this.getLs(this.userKey);
  }

  getUserToken() {
    return this.getLs(this.userKey);
  }

  setUserToken(value: string | Object) {
    return this.setLs(this.userKey, value);
  }

  getLs(key: string) {
    return localStorage.getItem(key);
  }

  setLs(key: string, value: string | Object) {
    return localStorage.setItem(
      key,
      typeof value === "string" ? value : JSON.stringify(value)
    );
  }
}
