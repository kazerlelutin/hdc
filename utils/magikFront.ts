import { Magic } from 'magic-sdk';

const m = ()=> new Magic(process.env.NEXT_PUBLIC_MAGIC_PUB_KEY)

export default m()
