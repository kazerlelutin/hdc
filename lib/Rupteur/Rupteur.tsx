import classes from "./Rupteur.module.css";
import { useEffect,useState } from 'react';
import { PREFIX } from '../../utils/constants';

type theme = "light" | "dark";

export default function Rupteur() {
    const 
        ls= PREFIX + "theme",    
        [theme, setTheme]= useState <theme> ('light');

    function handleClick() {
        const newTheme = theme === "light" ? "dark": "light";
        setTheme(newTheme);
        document.querySelector('body').setAttribute('data-theme',newTheme);
        localStorage.setItem(ls,newTheme)
    };

    useEffect(()=>{
        const saveTheme = localStorage.getItem(ls);
        if(saveTheme && (saveTheme === "light" || saveTheme === "dark")){
            document.querySelector('body').setAttribute('data-theme', saveTheme);
            setTheme(saveTheme)
        }
    },[]);

    return <div className={classes.container}>
        <div 
            className={classes.rupteur}
            data-testid="rupteur"
            data-theme={theme}
            onClick={handleClick}
        >
            <div className={classes.state}/>
        </div>
    </div>
}
