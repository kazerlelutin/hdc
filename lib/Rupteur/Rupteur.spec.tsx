import { render, cleanup, fireEvent, screen, waitFor } from '@testing-library/react';
import Rupteur from './Rupteur';
import { PREFIX } from '../../utils/constants';

describe('<Rupteur />', () => {

    afterEach(cleanup);
    it('Check snapshot', () => {
        const { asFragment } = render(<Rupteur/>);
        expect(asFragment()).toMatchSnapshot();
    });


    it('Check change with localStorage state',async () => {
        const 
            { getByTestId } = render(<Rupteur/>),
            ls = window.localStorage.getItem(PREFIX + "theme");
            
        expect(getByTestId('rupteur')).toHaveAttribute("data-theme");
        waitFor(()=>expect(ls).toEqual('light'));
    });

    it('if click on rupteur',async () => {
        const 
            { getByTestId } = render(<Rupteur/>),
            rupteur = getByTestId('rupteur'),
            body = document.querySelector('body');

        expect(getByTestId('rupteur')).toHaveAttribute("data-theme");
        expect(rupteur.getAttribute('data-theme')).toEqual('light');


        fireEvent.click(rupteur);

        expect(body.getAttribute('data-theme')).toEqual('dark');
        expect(rupteur.getAttribute('data-theme')).toEqual('dark');

        fireEvent.click(rupteur);
        expect(body.getAttribute('data-theme')).toEqual('light');
    });
});
