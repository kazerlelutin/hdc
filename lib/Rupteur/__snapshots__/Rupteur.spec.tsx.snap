// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rupteur /> Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="rupteur"
      data-testid="rupteur"
      data-theme="light"
    >
      <div
        class="state"
      />
    </div>
  </div>
</DocumentFragment>
`;
